src/Application.vue

```diff
@@ -138,151 +138,169 @@ export default {
+            setWidgetId(state.widgetData.widgetInstanceId)
+            setWidgetUUID(state.widgetData.widgetInstanceUUID)
+
+            // Check if there's a package name to open its component settings
+            if (typeof state.widgetData.widgetPackage !== 'undefined') {
+              let widgetPackage = state.widgetData.widgetLayout 
+                ? `${state.widgetData.widgetPackage}:${state.widgetData.widgetLayout}`
+                : state.widgetData.widgetPackage
+              this.appearanceGroup = _.find(state.activeTheme.settings.configuration, (config) => {
+                return config.packages && config.packages.indexOf(widgetPackage) > -1
+              })
+
+              // Set state to flag if widget has a flexbox parent
+              setParentFlex()
+
+              // Set state in widget mode
+              setWidgetMode(!!this.appearanceGroup)
+            }
 
-            // Set state to flag if widget has a flexbox parent
-            setParentFlex()
-            // Set state in widget mode
-            setWidgetMode(!!this.appearanceGroup)
-          }
+            // Set the active tab from widget data
+            this.isLoading = false
+            this.handleContextSwitch(tab)
+            this.handleAppearanceGroup(this.appearanceGroup)
 
-          // Set the active tab from widget data
-          this.isLoading = false
-          this.handleContextSwitch(tab)
-          this.handleAppearanceGroup(this.appearanceGroup)
+            return
+          }
 
-          return
-        }
+          // Set state to flag if widget has a flexbox parent to false
+          setParentFlex(false)
 
-        // Set state to flag if widget has a flexbox parent to false
-        setParentFlex(false)
-        // Set state in widget mode to false
-        setWidgetMode(false)
-        if (state.appearanceGroupOverlay.isOpen) {
-          closeAppearanceGroupSettings()
-        }
-        this.handleContextSwitch(tab)
-        this.isLoading = false
-      })
+          // Set state in widget mode to false
+          setWidgetMode(false)
 
-      Promise.all(themePromises)
-        .then(() => {
-          // Automatically create a theme instance if one doesn't exist
-          if (themeWithoutInstances !== themes.length) {
-            return
+          if (state.appearanceGroupOverlay.isOpen) {
+            closeAppearanceGroupSettings()
           }
 
-          const flipletTheme = _.find(themes, { name: FLIPLET_THEME })
+          this.handleContextSwitch(tab)
 
-          // Checks for older versions
-          ThemeModel.getAllVersions()
-            .then((result) => {
-              const allThemes = result.widgets
-              const versionOneTheme = _.find(allThemes, { name: 'Bootstrap', version: '1.0.0' })
+          this.isLoading = false
+        })
+    },
+    prepareToCreateDefaultInstance(options = {}) {
```

Comment:
I'd just call this:

```suggestion
    createDefaultInstance(options = {}) {
```

The other `createDefaultInstance()` function you created doesn't really do much more than calling one function, and isn't used repeatedly either.

The `prepareToX` functions don't really work well because it's effectively doing what it's preparing to do. It's like having `validateForm` as an event listener for form submission. You should have the function for what it's ultimately doing, e.g. `submitForm` or `saveForm` and then start the function within with a `validateX` function or similar if necessary.